remote-admin-create-cache/src/main/java/org/infinispan/tutorial/simple/remote/admin/InfinispanRemoteAdminCache.java

```diff
@@ -0,0 +1,98 @@
+        InfinispanRemoteAdminCache client = new InfinispanRemoteAdminCache();
+
+        client.createSimpleCache();
+        client.createCacheWithCacheConfiguration();
+        client.createCacheWithXMLConfiguration();
+
+        // Stop the client and release all resources
+        client.stop();
+    }
+
+    private void stop() {
+        manager.stop();
+    }
+
+    /**
+     * For this a configuration is needed within the infinispan.xml file!
+     * Add the following part to the <cache-container> element
+     * <code>
+     *    <distributed-cache-configuration name="MyDistCachecConfig" mode="ASYNC">
+     *      <state-transfer await-initial-transfer="false"/>
+     *     </distributed-cache-configuration>
+     * </code>
+     */
+    private void createCacheWithCacheConfiguration() {
+        manager.administration().getOrCreateCache("CacheWithConfigurationTemplate", "MyDistCachecConfig");
+        System.out.println("Cache with template exists or is created");
+
+        /* this cache is temporary and not persisted
+         * to see whether the cache is created start another instance and see that a log message will shown
+         * <code> [CLUSTER] [Context=TemporaryCacheWithConfigurationTemplate]ISPN100002: </code>
+         */
+        manager.administration().withFlags(AdminFlag.VOLATILE).getOrCreateCache("TemporaryCacheWithConfigurationTemplate", "MyDistCachecConfig");
+        System.out.println("Temporary cache with template exists or is created");
+    }
+
+    /**
+     * create a cache named CacheWithXMLConfiguration and provide the configuration
+     * having XML by using an implementation of BasicConfiguration.
+     * As result the cache is created and defaults are added.
+     */
+    private void createCacheWithXMLConfiguration() {
+        String cacheName = "CacheWithXMLConfiguration";
+        String xml = String.format("<infinispan>" +
+                                      "<cache-container>" +
+                                      "<distributed-cache name=\"%s\" mode=\"SYNC\">" +
+                                        "<locking isolation=\"READ_COMMITTED\"/>" +
+                                        "<transaction mode=\"NON_XA\"/>" +
+                                        "<expiration lifespan=\"60000\" interval=\"20000\"/>" +
+                                      "</distributed-cache>" +
+                                      "</cache-container>" +
+                                    "</infinispan>"
+                                    , cacheName);
+        manager.administration().getOrCreateCache(cacheName, new XMLStringConfiguration(xml));
+        System.out.println("Cache with configuration exists or is created");
+    }
+
+    /**
+     * create a simple cache without any configuration.
+     * It will use the server defaults.
+     */
+    private void createSimpleCache() {
+        try {
+            manager.administration().createCache("SimpleCache", (String)null);
```

Comment:
I would not put this method as an example. This API is just... not possible for me. It looks more an error than an API.
Is this used by users @wfink ? If 'not really', for me we should remove it from here IMO

This API of creating caches at runtime should be changed and I mentioned in the thread in zulip 
@tristantarrant @wburns

Reply:
I did not removed this.
Let's see whether it is a valid use or not

Reply:
we can change it afterwards